app/dao/templates_dao.py

```diff
@@ -56,3 +57,26 @@ def dao_get_template_versions(service_id, template_id):
     ).order_by(
         desc(TemplateHistory.version)
     ).all()
+
+
+def dao_get_templates_by_for_cache(cache):
+    if not cache or len(cache) == 0:
+        return []
+
+    # First create a subquery that is a union select of the cache values
+    # Then join templates to the subquery
+    cache_queries = [
+        db.session.query(sqlalchemy.sql.expression.bindparam("template_id" + str(i),
+                                                             uuid.UUID(template_id.decode())).label('template_id'),
```

Comment:
Very small but you can stick `from sqlalchemy import bindparam` at the top to make this a bit shorter.